sdk/core/core-rest-pipeline/review/core-rest-pipeline.api.md

```diff
@@ -300,6 +303,28 @@ export interface RestErrorOptions {
     statusCode?: number;
 }
 
+// @public
+export function retryPolicy(...strategies: RetryStrategy[]): PipelinePolicy;
+
+// @public
+export interface RetryStrategy {
+    logger?: AzureLogger;
+    meetsConditions?(state: RetryStrategyState): boolean;
+    name: string;
+    updateRetryState(state: RetryStrategyState): RetryStrategyState;
+}
+
+// @public
+export interface RetryStrategyState {
+    maxRetries?: number;
+    redirectTo?: string;
+    response?: PipelineResponse;
+    responseError?: RestError;
+    retryAfterInMs?: number;
+    retryCount: number;
+    throwError?: RestError;
+}
+
```

Comment:
This section of the public API changes will help us migrate other retry strategies in the repository, like the storage’s retry policy.